core/src/main/scala/org/bitcoins/core/crypto/Schnorr.scala

```diff
@@ -0,0 +1,71 @@
+package org.bitcoins.core.crypto
+
+import org.bitcoin.NativeSecp256k1
+import scodec.bits.ByteVector
+
+object Schnorr {
+
+  def sign(
+      msg: ByteVector,
+      privateKey: ECPrivateKey): SchnorrDigitalSignature = {
+    if (msg.length != 32) {
+      throw new IllegalArgumentException(s"Message must be 32 bytes, got $msg")
+    } else {
+      val sigArr =
+        NativeSecp256k1.schnorrSign(msg.toArray, privateKey.bytes.toArray)
+
+      SchnorrDigitalSignature.fromBytes(ByteVector(sigArr))
+    }
+  }
+
+  def signWithNonce(
+      msg: ByteVector,
+      privateKey: ECPrivateKey,
+      nonce: ECPrivateKey): SchnorrDigitalSignature = {
```

Comment:
`nonce` here is a priv key because it needs to be 32 bytes, right? Not sure how I feel about using `ECPrivateKey`, as types are also very important to convey meaning to params etc